src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs

```diff
@@ -60,7 +61,7 @@ internal sealed class LocalFunctionSymbol : SourceMethodSymbolWithAttributes
 
             if (syntax.TypeParameterList != null)
             {
-                binder = new WithMethodTypeParametersBinder(this, binder);
+                _withTypeParamsBinder = new WithMethodTypeParametersBinder(this, binder);
```

Comment:
>_withTypeParamsBinder [](http://example.com/codeflow?start=16&length=21)

It looks like this field is unnecessary. We can create ```WithMethodTypeParametersBinder``` as needed and it doesn't look like we have to cache it.